README.md

```diff
@@ -30,25 +54,225 @@ npm install --save ipfsd-ctl
 
 IPFS daemons are already easy to start and stop, but this module is here to do it from JavaScript itself.
 
+### Local node
+
 ```js
 // Start a disposable node, and get access to the api
-// print the node id, and kill the temporary daemon
+// print the node id, and stop the temporary daemon
 
 // IPFS_PATH will point to /tmp/ipfs_***** and will be
 // cleaned up when the process exits.
 
-var ipfsd = require('ipfsd-ctl')
+const daemonFactory = require('ipfsd-ctl')
+const local = daemonFactory.localController
```

Comment:
Thinking a little bit more about this. Why does the user need to specify a local/remote controller this way?

Wouldn't it be more simpler if there was a setup step:

```
const DaemonFactory = require('ipfsd-ctl')
const df = new DaemonFactory(options)
```

Where default options is to:

- Do local for Node.js
- Do remote with default endpoint for Browser.

Reply:
I thought about doing that at some point, but I settled on not making any assumptions on behalf of the user. I think this is still a valid discussion point, but my reasoning was basically that - let the user choose which controller it wants to use.

Reply:
The user can still choose. I'm not particularly liking this `daemonFactory.localController`. API, I feel that the df should be an instance of a DaemonFactory and the user can pick their set up through options.

Reply:
The one drawback with this approach is that it forces the DF to be stateful rathern than stateless as in the case of simple "static" methods. How about a middle ground? A factory method that will return the correct controller factory? Something like (pseudocode):

```js
class DaemonFactory {
  static createFactory(options): return ControllerFactory
}
```

where `ControllerFactory` implements the `spaw` interface? Note that it doesn't have to be a class, can just as well be a plain old `module.exports = function(){}`.

Reply:
Yeah, I like `DaemonFactory.create` too 👍   (no need for the Factory word again)